OpenRA.Renderer.Sdl2/Sdl2GraphicsDevice.cs

```diff
@@ -330,6 +330,43 @@ public void SetLineWidth(float width)
 			ErrorHandler.CheckGlError();
 		}
 
+		public Bitmap TakeScreenshot()
+		{
+			var rect = new Rectangle(Point.Empty, size);
+			var bitmap = new Bitmap(rect.Width, rect.Height, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+			var data = bitmap.LockBits(rect,
+				System.Drawing.Imaging.ImageLockMode.WriteOnly, System.Drawing.Imaging.PixelFormat.Format32bppArgb);
+
+			try
```

Comment:
Not handling exceptions, handling the data that is being unlocked here (is what @pchote meant).
And even if we catch an exception here, we have no way to "handle" it. The best thing to do is keep quiet. Hence no catching is needed.
But we do want the finally so we can unlock the data, otherwise we're returning locked data a few lines below, which might cause problems later.